sdk/apis/workload/v1alpha1/types.go

```diff
@@ -188,4 +188,15 @@ const (
 	// and further work should be done to define such (up)syncing strategies at a more appropriate level
 	// (SyncTarget, KCP namespace, KCP workspace ?).
 	ExperimentalUpsyncDerivedResourcesAnnotationKey = "experimental.workload.kcp.io/upsync-derived-resources"
+
+	// InternalWorkspaceURLAnnotationKey is an annotation dynamically added on resources exposed
+	// by the Syncer Virtual Workspace to be synced by the Syncer.
+	// It contains the external URL of the workspace the resource is part of.
```

Comment:
Moving workspaces between shards should be O(1) or as close as we can get to it. Having to batch-mutate N objects is a no-go I think.

Reply:
Nothing is batch-mutated here.
This annotation is added "on-the-fly" by the Syncer virtual workspace when exposing resources to the Syncer.